### PR TITLE
ISPN-9705 and ISPN-9706 Null values bugs

### PR DESCRIPTION
**Pull request by karesti**
https://issues.jboss.org/browse/ISPN-9705
https://issues.jboss.org/browse/ISPN-9706

both have to be backported

### CODE REVIEW

File: spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheTest.java [hunk 1]
```diff
@@ -266,6 +266,28 @@ public void testValueLoaderWithExceptionWhenLoading() {
       cache.get("test", () -> {throw new IllegalStateException();});
    }
 
+   @Test
+   public void testPutNullAndGetWithClass() {
+      //when//then
```
Comment (danberindei) [inline spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheTest.java]: Is this really useful?

File: spring/spring4/spring4-common/src/main/java/org/infinispan/spring/provider/SpringCache.java [hunk 2]
```diff
@@ -113,7 +115,7 @@ public ValueWrapper get(final Object key) {
                   T newValue = valueLoader.call();
                   // we can't use computeIfAbsent here since in distributed embedded scenario we would
                   // send a lambda to other nodes. This is the behavior we want to avoid.
-                  value = (T) nativeCache.putIfAbsent(key, newValue);
+                  value = (T) nativeCache.putIfAbsent(key, newValue == null ? NullValue.NULL: newValue);
```
Comment (danberindei) [inline spring/spring4/spring4-common/src/main/java/org/infinispan/spring/provider/SpringCache.java]:
```suggestion
                  value = (T) nativeCache.putIfAbsent(key, newValue == null ? NullValue.NULL : newValue);
```

File: spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheTest.java (hunk 1 above)
Comment (karesti) [reply spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheTest.java]: Both of three tests were different errors

**Comment by karesti** [comment]
@danberindei I reported the checkstyle modification

File: spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheTest.java (hunk 1 above)
Comment (danberindei) [reply spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/provider/SpringCacheTest.java]: I meant the comment, IMO it doesn't add anything

**Comment by karesti** [comment]
Well it should be deleted everywhere tbh

Le lun. 19 nov. 2018 à 16:30, Dan Berindei <notifications@github.com> a
écrit :

**Comment by danberindei** [comment]
Integrated, thanks Katia!

**Pull request merged**